### PR TITLE
Fix for track using assembly alias not displaying

### PR DESCRIPTION
**Pull request by garrettjstevens**
Found this as part of other work, but am submitting it separately since it's a self-contained bug fix.

In the case where e.g. as assembly had a `name` of `"hg19"` and `aliases` of `["GRCh37"]`, if a track configuration used `"assemblyNames": ["GRCh37"]`, the track would not show up in the track selector, even though it was using a valid assembly alias.

Fixing that then revealed that the track would not display in the LinearGenomeView because the check comparing the track and displayedRegion assembly names was not being performed properly.

Fixes both bugs and adds a test to confirm the proper behavior.

### CODE REVIEW

**Comment by cmdcolin** [comment]
I noticed that both the display and the assembly names checks were checking for an array intersection in different ways, so I just codified both as performing an array intersection test...just pushed to a branch here https://github.com/GMOD/jbrowse-components/tree/fix_assembly_alias_mod

File: plugins/data-management/src/HierarchicalTrackSelectorWidget/model.js [hunk 1]
```diff
@@ -56,13 +56,23 @@ export default pluginManager =>
           return []
         }
 
+        const session = getSession(self)
+        const { assemblyManager } = session
+        const assembly = assemblyManager.get(assemblyName)
```
Comment (cmdcolin) [inline plugins/data-management/src/HierarchicalTrackSelectorWidget/model.js:61]: why is the check for initialized needed now after this PR?

File: products/jbrowse-web/src/tests/JBrowse.test.js [hunk 2]
```diff
@@ -96,6 +97,32 @@ test('variant track test - opens feature detail view', async () => {
...
+    const pluginManager = getPluginManager({
+      ...volvoxConfigSnapshot,
+      tracks: [assemblyAliasVariantTrack],
+    })
+    const state = pluginManager.rootModel
+    const { findByTestId, findByText } = render(
+      <JBrowse pluginManager={pluginManager} />,
+    )
+    await findByText('Help')
+    state.session.views[0].setNewView(0.05, 5000)
```
Comment (cmdcolin) [inline products/jbrowse-web/src/tests/JBrowse.test.js:116]: +1 for test :)

File: plugins/data-management/src/HierarchicalTrackSelectorWidget/model.js (hunk 1 above)
Comment (garrettjstevens) [reply plugins/data-management/src/HierarchicalTrackSelectorWidget/model.js:62]: Good point, it's not actually needed. It just needs to check if `assembly` is defined. I've removed it.

**Comment by garrettjstevens** [comment]
@cmdcolin Thanks! I've merged in that change.

File: plugins/linear-genome-view/src/BaseLinearDisplay/models/serverSideRenderedBlock.ts [hunk 3]
```diff
@@ -189,7 +189,11 @@ function renderBlockData(self: Instance<BlockStateModel>) {
       let matchFound = false
       assemblyNames.forEach((assemblyName: string) => {
         const assembly = assemblyManager.get(assemblyName)
-        if (assembly && assembly.aliases.includes(self.region.assemblyName))
+        if (
+          assembly &&
+          (assembly.name === self.region.assemblyName ||
+            assembly.aliases.includes(self.region.assemblyName))
+        )
```
Comment (rbuels) [inline plugins/linear-genome-view/src/BaseLinearDisplay/models/serverSideRenderedBlock.ts:196]: Is assembly a proper object here?  Could we make a method on assemblies called something like `hasName`, so this check would be `assembly && assembly.hasName(self.region.assemblyName)`?

Comment (rbuels) [reply plugins/linear-genome-view/src/BaseLinearDisplay/models/serverSideRenderedBlock.ts:196]: I bet there are other places in the code where this check is being done, probably want to grep for those

Comment (cmdcolin) [reply plugins/linear-genome-view/src/BaseLinearDisplay/models/serverSideRenderedBlock.ts:196]: Could do assembly.allAlliases.includes(name) or make helper method

**Pull request merged**